--- Rebuild the Cube.sql	Tue Sep 17 11:10:38 2019
+++ Rebuild the Cube.sql	Tue Sep 17 11:10:38 2019
@@ -167 +167,3 @@
-                            'EDU'
+                            'EDU',
+                            'DYNAC',
+                            'RESTOCK'
@@ -436,0 +439 @@
+           AND  custinfo.corporate_form != 'KEY'
@@ -502 +505,3 @@
-                        'EDU'
+                        'EDU',
+                        'DYNAC',
+                        'RESTOCK'
@@ -807,0 +813 @@
+          AND custinfo.corporate_form != 'KEY'
@@ -849 +855,3 @@
-                        'EDU'
+                        'EDU',
+                        'DYNAC',
+                        'RESTOCK'
@@ -1071,0 +1080 @@
+          AND  custinfo.corporate_form != 'KEY'
@@ -1118 +1127,3 @@
-                        'EDU'
+                        'EDU',
+                        'DYNAC',
+                        'RESTOCK'
@@ -1346,0 +1358 @@
+          AND  custinfo.corporate_form != 'KEY'
@@ -1388 +1400,3 @@
-                        'EDU'
+                        'EDU',
+                        'DYNAC',
+                        'RESTOCK'
@@ -1610,0 +1625 @@
+          AND  custinfo.corporate_form != 'KEY'
@@ -1660 +1675,3 @@
-                        'EDU'
+                        'EDU',
+                        'DYNAC',
+                        'RESTOCK'
@@ -1890,0 +1908 @@
+          AND  custinfo.corporate_form != 'KEY'
@@ -1956 +1974,3 @@
-                        'EDU'
+                        'EDU',
+                        'DYNAC',
+                        'RESTOCK'
@@ -2177,0 +2198,2 @@
+        WHERE   custinfo.corporate_form != 'KEY'
+        
@@ -2451,0 +2474,12 @@
+        WHERE   invhead.salesmngr NOT IN 
+                    (
+                        'DORIT AVIV',
+                        'MARKETING',
+                        'R' || CHR(38) || 'D'
+                    )
+          AND   invhead.customerno NOT IN 
+                    (
+                        'Cust1002664',
+                        'Cust1002656',
+                        'Cust1002224'
+                    )    
@@ -2703 +2736,0 @@
-    ;
@@ -2704,0 +2738,3 @@
+        WHERE   custinfo.corporate_form != 'KEY'
+          AND   invhead.customerno != 'Cust1002028';
+    
